### PR TITLE
[monitoring] Use aiohttp.AppKey to add static typing to Application storage

### PR DESCRIPTION
**Pull request by daniel-goldstein**
Forgive my throwing the dev proxy change in here too, but this gets us static typechecking on stuff stored in `app` across the codebase except for `batch`. That's a beast that will be a lot of changes to port to `AppKey`s and I don't want to mess with the current in flight PRs.

### CODE REVIEW

File: devbin/dev_proxy.py [hunk 1]
```diff
@@ -50,12 +50,15 @@ async def render_html(request: web.Request, context: dict):
...
+
+
 async def on_startup(app: web.Application):
-    app['backend_client'] = Session(credentials=hail_credentials())
+    app[BC] = Session(credentials=hail_credentials())
 
 
 async def on_cleanup(app: web.Application):
-    await app['backend_client'].close()
+    await app[AppKeys.BC].close()
```
Comment (iris-garden) [inline devbin/dev_proxy.py:61]:
```suggestion
    await app[BC].close()
```
i think this is meant to just refer to the variable in this file, not a member of `AppKeys`, right?

File: devbin/dev_proxy.py [hunk 2]
```diff
@@ -50,12 +50,15 @@ async def render_html(request: web.Request, context: dict):
     return await render_template(SERVICE, request, **context, cookie_domain='localhost:8000')
 
 
+BC = web.AppKey('backend_client', Session)
```
Comment (iris-garden) [inline devbin/dev_proxy.py:53]: can we move the variable definition above the `proxy` function, since that uses it too?

**Review by iris-garden — CHANGES_REQUESTED** [review]
couple comments on the devbin script, otherwise lgtm!

File: devbin/dev_proxy.py (hunk 1 above)
Comment (daniel-goldstein) [reply devbin/dev_proxy.py:61]: Good catch

File: devbin/dev_proxy.py (hunk 2 above)
Comment (daniel-goldstein) [reply devbin/dev_proxy.py:53]: yup!

**Review by iris-garden — APPROVED** [review]

**Pull request merged**